apis/run/v1beta1/job_types.go

```diff
@@ -0,0 +1,800 @@
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Optional. User-provided annotations, which are stored in GCP.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Optional. Settings for Binary Authorization feature.
+	BinaryAuthorization *BinaryAuthorization `json:"binaryAuthorization,omitempty"`
+
+	// Optional. Arbitrary identifier for the API client.
+	Client *string `json:"client,omitempty"`
+
+	// Optional. Arbitrary version identifier for the API client.
+	ClientVersion *string `json:"clientVersion,omitempty"`
+
+	// Optional. The launch stage of the job.
+	// Possible values are `LAUNCH_STAGE_UNSPECIFIED`, `UNIMPLEMENTED`, `PRELAUNCH`, `EARLY_ACCESS`, `ALPHA`, `BETA`, `GA`, `DEPRECATED`.
+	LaunchStage *string `json:"launchStage,omitempty"`
+
+	// Required. The template used to create executions for this Job.
+	Template *ExecutionTemplate `json:"template"`
+}
+
+// RunJobStatus defines the config connector machine state of RunJob
+type RunJobStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the RunJob resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *RunJobObservedState `json:",inline"`
+}
+
+// +kcc:spec:proto=google.cloud.run.v2.Job
+type RunJobObservedState struct {
+
+	// Output only. The creation time.
+	CreateTime *string `json:"createTime,omitempty"`
+
+	// Output only. The last-modified time.
+	UpdateTime *string `json:"updateTime,omitempty"`
+
+	// Output only. The deletion time. It is only populated as a response to a
+	// Delete request.
+	DeleteTime *string `json:"deleteTime,omitempty"`
+
+	// Output only. Email address of the authenticated creator.
+	Creator *string `json:"creator,omitempty"`
+
+	// Output only. Number of executions created for this job.
+	ExecutionCount *int32 `json:"executionCount,omitempty"`
+
+	// Output only. A system-generated fingerprint for this version of the
+	// resource. May be used to detect modification conflict during updates.
+	Etag *string `json:"etag,omitempty"`
+
+	// Output only. For a deleted resource, the time after which it will be
+	// permamently deleted.
+	ExpireTime *string `json:"expireTime,omitempty"`
+
+	// Output only. Email address of the last authenticated modifier.
+	LastModifier *string `json:"lastModifier,omitempty"`
+
+	// Note: ExecutionReference in Run v2 proto uses a single object, but legacy KCC CRD uses an array
```

Comment:
Call out for the exception for `LatestCreatedExecution`. this results in the manual change in mapper.

Reply:
yes!